software/main.py

```diff
@@ -0,0 +1,266 @@
+                            
+                        elif abs(basket.x-reso_x_mid)>30:
+                            #orbit_speeds = Orbit(radius, reso_x_mid, xcord, dist)
+                             
+                            speedY=0
+                            speedX=0.5
+                            speedR=1000*speedX/(640-radius)
+
+                            if radius > (dist-5):
+                                speedY += (radius-dist)/100
+                            elif radius > (dist+5):
+                                speedY += (radius-dist)/100
+                            
+                            
+                            if xcord > (reso_x_mid + 1) or xcord < (reso_x_mid - 1):
+                                speedR += (reso_x_mid- xcord) / 100 
+                                                   
+                            #robot.move(orbit_speeds[0],orbit_speeds[1],orbit_speeds[2])
+                            robot.move(speedX, speedR, speedY)                        
+                    else:
+                        #Orbit
+                        #orbit_speeds = Orbit(radius, reso_x_mid, xcord, dist)
+
+                        speedY=0
+                        speedX=0.5
+                        speedR=1000*speedX/(640-radius)
+
+                        if radius > (dist-5):
+                            speedY += (radius-dist)/100
+                        elif radius > (dist+5):
+                            speedY += (radius-dist)/100
+                        
+                        
+                        if xcord > (reso_x_mid + 1) or xcord < (reso_x_mid - 1):
+                            speedR += (reso_x_mid- xcord) / 100 
+                                                
+                        #robot.move(orbit_speeds[0],orbit_speeds[1],orbit_speeds[2])
+                        robot.move(speedX, speedR, speedY)
+                        
+                        #robot.move(orbit_speeds[0],orbit_speeds[1],orbit_speeds[2])
+            
+                            
+                    
+
+            elif state == "throw":
+                #calc speed in m/s
+
+                processedData.balls.sort(key= lambda x: x.size)
+                targeted_ball=processedData.balls[-1]
+                dist=targeted_ball.distance
+                delta=((xcord-reso_x_mid)/reso_x_mid)
+                
+
+                
+                #Moves to ball
+                speedY = 0.3
+                if dist > 400:
+                    speedR = 0
+                    speedX = 0
+                    speedT=950
+                    baskety=processedData.basket_m.y
+                    for i in range(len(tsize)):
+                        if baskety>=tsize[i]:
+                            speedT=((-500+baskety)*tspeeds[i])/(-500+tsize[i])
+                            print("Throwing with speed: ", speedT)
+                            break
+                    robot.throw(speedX, speedR, speedY, int(speedT))

```

Comment:
Instead of sleeping you should continue correcting the aim of the robot. You might not drive straight or a opponent might bump you when throwing, throwing off your aim

Reply:
In future. For now it works because it goes to sleep when ball is very close to thrower